types/base.d.ts

```diff
@@ -120,5 +120,13 @@ export interface ContextualCallbackTestContext extends CallbackTestContext {
 	context: any;
 }
 
+export interface Macro<T, I, E> {
+	(t: T, input: I, expected: E): void;
+	title? (providedTitle: string, input: I, expected: E): string;
+}
+export type Macros<T, I, E> = Macro<T, I, E> | Macro<T, I, E>[];
+
 export function test(name: string, run: ContextualTest): void;
 export function test(run: ContextualTest): void;
+export function test<I, E> (name: string, run: Macros<ContextualTestContext, I, E>, input: I, expected: E): void;
+export function test<I, E> (run: Macros<ContextualTestContext, I, E>, input: I, expected: E): void;
```

Comment:
That would require Microsoft/TypeScript#5453. That's not implemented yet, so for now these are the options:

- Use `...args: any[]` - no type checking at all
- Support only two arguments (current)
- Support max two arguments, and mark them optional (`?`) - this means that the amount of arguments isn't checked; a macro with 2 arguments can be run without arguments, and the other way around
- Support zero, one or two arguments

Using `...args: any[]` means that TypeScript doesn't check anything at all, so that's not a good idea in my opinion. The last option means that three interfaces are needed (`Macro0`, `Macro1`, `Macro2`), and 3 x 2 = 6 function overloads for Macros. So that's not ideal, but it might be the best solution. (That option could be extended to any amount of arguments of course. Since the definition is already generated, it wouldn't be very hard to add.)